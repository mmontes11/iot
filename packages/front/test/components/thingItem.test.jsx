import React from "react";
import { shallow } from "enzyme";
import ThingItem from "components/thingItem";

describe("components/thingItem", () => {
  const onClick = jest.fn();
  const wrapper = shallow(<ThingItem name="foo" isSelected={false} onClick={onClick} />);
  it("renders when it's not selected", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders when it's selected", () => {
    const wrapperSelected = shallow(<ThingItem name="bar" isSelected onClick={() => undefined} />);
    expect(wrapperSelected).toMatchSnapshot();
  });
  it("simulates a click", () => {
    wrapper
      .find("div")
      .first()
      .simulate("click");
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
