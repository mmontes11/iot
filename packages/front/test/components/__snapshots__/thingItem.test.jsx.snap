// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/thingItem renders when it's not selected 1`] = `
<div
  className="thing-item-card"
  onClick={[MockFunction]}
  onKeyPress={[MockFunction]}
  role="button"
  tabIndex={0}
>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <p
        className="title is-6"
      >
        foo
      </p>
    </div>
  </div>
</div>
`;

exports[`components/thingItem renders when it's selected 1`] = `
<div
  className="thing-item-card is-selected"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <p
        className="title is-6"
      >
        bar
      </p>
    </div>
  </div>
</div>
`;
